component/src/main/java/io/siddhi/extension/store/mongodb/MongoSelectExpressionVisitor.java

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.siddhi.extension.store.mongodb;
+
+import io.siddhi.core.table.record.BaseExpressionVisitor;
+import io.siddhi.extension.store.mongodb.exception.MongoTableException;
+import io.siddhi.extension.store.mongodb.util.MongoTableUtils;
+import io.siddhi.query.api.definition.Attribute;
+import io.siddhi.query.api.expression.condition.Compare;
+
+import java.util.Arrays;
+import java.util.Stack;
+
+/**
+ * Class representing MongoDB select attribute condition implementation.
+ */
+public class MongoSelectExpressionVisitor extends BaseExpressionVisitor {
+
+    private Stack<String> conditionOperands;
+    private int streamVarCount;
+    private int constantCount;
+    private String[] supportedFunctions = {"sum", "avg", "min", "max", "count"};
+    private boolean isAttributeFunctionUsed;
+    private boolean isCountFunction;
+    private boolean isNullCheck;
+
+    public MongoSelectExpressionVisitor() {
+        this.conditionOperands = new Stack<>();
```

Comment:
Never initialize generalised objects
```suggestion
        this.conditionOperands = new Stack<String>();
```

Reply:
Fixed in
https://github.com/siddhi-io/siddhi-store-mongodb/pull/56/commits/94fba768238670966474ed6ac8a757455790895b